client/providers/discover/discover.go

```diff
@@ -33,7 +33,7 @@ func NewPublicKeyRecord(key jwk.Key, issuer string) (*PublicKeyRecord, error) {
 	if key.Algorithm() == jwa.ES256 {
 		pubKey = new(ecdsa.PublicKey)
 	}
-	err := key.Raw(pubKey)
+	err := key.Raw(&pubKey)
```

Comment:
```suggestion
    if key.Algorithm() ==nil {
        // OPs such as azure (microsoft) does not specify alg in their JWKS. To handle this case 
        // assume no alg in JWKS means RSA. as OIDC says OPs must use RSA 
        pubKey = new(rsa.PublicKey)
	}
    err := key.Raw(&pubKey)
```

Reply:
Pushed a0bfa92 and added unittest to `discover_test.go`. However, is it fair to assume if alg is not specified this means RSA and `RS256`? What if it means RSA but sig alg is `RS384` or `RS512`? OIDC spec says OP must use RSA but how do we know if missing alg actually means the provided key is using RSA, or are we just assuming that?

Reply:
Ok. After further examination in the OIDC spec, it does say that `RS256` is the default, so this seems fine to assume the alg as well.

> The alg value SHOULD be the default of RS256 or the algorithm sent by the Client in the id_token_signed_response_alg parameter during Registration.

Reply:
The OIDC spec says:
> OPs MUST support signing ID Tokens with the RSA SHA-256 algorithm (an alg value of RS256), unless the OP only supports returning ID Tokens from the Token Endpoint (as is the case for the Authorization Code Flow) and only allows Clients to register specifying none as the requested ID Token signing algorithm. 

https://openid.net/specs/openid-connect-core-1_0.html#ServerMTI

So RS256 is the requirement not just RSA. That said, I like that you are thinking about this and it is something we should consider, not all OPs follow the spec [and other projects have run into issues with OPs signing ID Tokens with RS512.](https://community.auth0.com/t/getting-rs512-error-when-using-openid-connect/68579)

Reply:
Jinks!